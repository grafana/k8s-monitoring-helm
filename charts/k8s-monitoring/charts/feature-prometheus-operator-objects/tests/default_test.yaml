# yamllint disable rule:document-start rule:line-length rule:trailing-spaces
suite: Test default values
templates:
  - configmap.yaml
tests:
  - it: should create a ConfigMap
    set:
      deployAsConfigMap: true
    asserts:
      - isKind:
          of: ConfigMap
      - equal:
          path: data["module.alloy"]
          value: |-
            declare "prometheus_operator_objects" {
              argument "metrics_destinations" {
                comment = "Must be a list of metric destinations where collected metrics should be forwarded to"
              }
            
              // Prometheus Operator PodMonitor objects
              prometheus.operator.podmonitors "pod_monitors" {
                clustering {
                  enabled = true
                }
                scrape {
                  default_scrape_interval = "60s"
                }
                forward_to = argument.metrics_destinations.value
              }
            
              // Prometheus Operator Probe objects
              prometheus.operator.probes "pod_monitors" {
                clustering {
                  enabled = true
                }
                scrape {
                  default_scrape_interval = "60s"
                }
                forward_to = argument.metrics_destinations.value
              }
            
              // Prometheus Operator ServiceMonitor objects
              prometheus.operator.servicemonitors "service_monitors" {
                clustering {
                  enabled = true
                }
                scrape {
                  default_scrape_interval = "60s"
                }
                forward_to = argument.metrics_destinations.value
              }
            }
