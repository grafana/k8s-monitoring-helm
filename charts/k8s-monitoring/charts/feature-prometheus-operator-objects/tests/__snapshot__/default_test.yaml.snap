should generate the default configuration:
  1: |
    |-
      declare "prometheus_operator_objects" {
        argument "metrics_destinations" {
          comment = "Must be a list of metric destinations where collected metrics should be forwarded to"
        }

        // Prometheus Operator PodMonitor objects
        prometheus.operator.podmonitors "pod_monitors" {
          clustering {
            enabled = true
          }
          scrape {
            default_scrape_interval = "60s"
          }
          forward_to = argument.metrics_destinations.value
        }

        // Prometheus Operator Probe objects
        prometheus.operator.probes "probes" {
          clustering {
            enabled = true
          }
          scrape {
            default_scrape_interval = "60s"
          }
          forward_to = argument.metrics_destinations.value
        }

        // Prometheus Operator ServiceMonitor objects
        prometheus.operator.servicemonitors "service_monitors" {
          clustering {
            enabled = true
          }
          scrape {
            default_scrape_interval = "60s"
          }
          forward_to = argument.metrics_destinations.value
        }
      }
